derp/derphttp/websocket.go

```diff
@@ -18,9 +20,14 @@ func init() {
 	dialWebsocketFunc = dialWebsocket
 }
 
-func dialWebsocket(ctx context.Context, urlStr string) (net.Conn, error) {
+func dialWebsocket(ctx context.Context, urlStr string, tlsConfig *tls.Config) (net.Conn, error) {
 	c, res, err := websocket.Dial(ctx, urlStr, &websocket.DialOptions{
 		Subprotocols: []string{"derp"},
+		HTTPClient: &http.Client{
```

Comment:
This isn't available in `js`, so we might want to move this assignment out to a separate file behind build tag. Maybe a function like `wsDialOptionsWithHTTPClient(opts, tlsConfig)`.

Reply:
Good point! Will fix.

Reply:
Check now!